### PR TITLE
Move data tier roles to server

### PR DESCRIPTION
**Pull request by jasontedor**
This commit moves the data tier roles to server. It is no longer necessary to separate these roles from server as we no longer build distributions that would not contain these roles. Moving these roles will simplify many things. This is deliberately the smallest possible commit that moves these roles. Other aspects related to the data tiers can move in separate, also small, commits.

Relates #71013

### CODE REVIEW

**Review by dakrone — APPROVED** [review]
LGTM assuming CI is happy

**Comment by jasontedor** [comment]
@dakrone I pushed a production code change to 4a733745a507593cd6279d42d8d7e8ab25d2750b (for simplification, and correctness), can you please review. There was a failing test in `DataTierTests#testNodeSelection` that failed prior to 4a733745a507593cd6279d42d8d7e8ab25d2750b and now passes, and I believe this change is future proof to us adding new roles.

**Comment by dakrone** [comment]
This reproduces a failure for me:

File: server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java [hunk 1]
```diff
@@ -350,29 +353,31 @@ public DiscoveryNode resolveNode(String node) {
...
-                        } else if (DiscoveryNodeRole.INGEST_ROLE.roleName().equals(matchAttrName)) {
+                            final Function<String, Boolean> mutation;
                             if (Booleans.parseBoolean(matchAttrValue, true)) {
-                                resolvedNodesIds.addAll(ingestNodes.keys());
+                                mutation = resolvedNodesIds::add;
                             } else {
-                                resolvedNodesIds.removeAll(ingestNodes.keys());
+                                mutation = resolvedNodesIds::remove;
                             }
-                        } else if (DiscoveryNode.COORDINATING_ONLY.equals(matchAttrName)) {
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java:371]: I believe this case is no longer handled. `COORDINATING_ONLY` is not included in `BUILTIN_ROLES`, but is allowed in node specifications.

`DiscoveryNodesTests.testCoordinatorOnlyNodes` fails like 1/3 of the time.

**Review by henningandersen — APPROVED** [review]
LGTM, thanks for taking care of this.

**Pull request merged**